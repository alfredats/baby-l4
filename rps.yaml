
rules: 
query: 
data:
- name: Game
  attributes:
  FieldDecl ErrT (FldNm "participants") (TupleT [ClassT (ClsNm "Player"),ClassT (ClsNm "Player")])
  FieldDecl ErrT (FldNm "winner") (ClassT (ClsNm "Player"))
terms: 