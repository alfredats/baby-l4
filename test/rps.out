rules: rps.pl
query: winner(X,Y)
data:
  - name: game
    type: String
    minimum: 0
    encodings:
      - game(X)
    attributes:
    - name: Player
      type: String
      exactly: 2
      encodings:
        - player(X)
        - participant(X,Y)
      attributes:
      - name: throw
        type: Enum
        options:
          rock: Rock
          paper: Paper
          scissors: Scissors
        encodings:
          - throw(Y,X)
terms:
  - winner: |
      According to rule 4 of Rock Paper Scissors, the winner
      is the player in the game whose throw beats the other player
      in the game.
